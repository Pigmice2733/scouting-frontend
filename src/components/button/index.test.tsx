import { h } from 'preact'
import testUtils from 'test-utils'
import Button from '.'

test('triggers cb', () => {
  const cb = jest.fn()
  const button = testUtils<HTMLButtonElement>(<Button onClick={cb} />)
  expect(cb).not.toHaveBeenCalled()
  button.click()
  expect(cb).toHaveBeenCalled()
})

test('uses an `a` for `href`', () => {
  expect(testUtils(<Button href="google.com">Hello</Button>)).toMatchSnapshot()
})

test('uses an `button` otherwise', () => {
  expect(
    testUtils(
      <Button type="foo" value="hello" disabled>
        Hello
      </Button>
    )
  ).toMatchSnapshot()
})
