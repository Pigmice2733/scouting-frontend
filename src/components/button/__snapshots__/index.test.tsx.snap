// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses an \`a\` for \`href\` 1`] = `
<a
  class="button "
  href="google.com"
>
  Hello
</a>
`;

exports[`uses an \`button\` otherwise 1`] = `
<button
  class="button "
  disabled=""
  type="foo"
  value="hello"
>
  Hello
</button>
`;
